pudb/debugger.py

```diff
@@ -2838,11 +2841,16 @@ def interaction(self, exc_tuple, show_exc_dialog=True):
                 (None, " "),
                 ("header warning", "[POST-MORTEM MODE]")
                 ])
+            self.show_header()
+            CONFIG["hide_header"] = False
+
         elif exc_tuple is not None:
             caption.extend([
                 (None, " "),
                 ("header warning", "[PROCESSING EXCEPTION - hit 'e' to examine]")
                 ])
+            self.show_header()
+            CONFIG["hide_header"] = False
```

Comment:
This is a little weird; at this point the on-disk configuration will start disagreeing from the in-memory one. I'm guessing if one opens the preferences dialog, this changed config will then be persisted to disk, but not otherwise, which may be confusing.

Overall, I find this code a bit too stateful. I would like it better if there were simply one place that decides whether the header is shown (if configured so, or if an there's an exception), and then all the places that may cause state changes affecting this decision call that central place.